cyclonedx-bom/src/models/bom.rs

```diff
@@ -128,8 +128,12 @@ impl Bom {
     pub fn parse_from_json<R: std::io::Read>(
         mut reader: R,
     ) -> Result<Self, crate::errors::JsonReadError> {
-        let json: serde_json::Value = serde_json::from_reader(&mut reader)?;
+        Self::parse_json_value_value(serde_json::from_reader(&mut reader)?)
+    }
 
+    /// General function to parse a pre-parsed JSON file, fetches the `specVersion` field first,
+    /// then applies the right conversion.
+    pub fn parse_json_value_value(json: Value) -> Result<Self, crate::errors::JsonReadError> {
```

Comment:
I like the basic direction. But shouldn't this be called `parse_json_value` rather than `parse_json_value_value`?

Reply:
Doppelmoppel :grimacing: … fixed it.